builder/vmware/common/step_compact_disk.go

```diff
@@ -26,28 +28,46 @@ type StepCompactDisk struct {
 func (s StepCompactDisk) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
 	driver := state.Get("driver").(Driver)
 	ui := state.Get("ui").(packer.Ui)
-	full_disk_path := state.Get("full_disk_path").(string)
+	diskPaths := state.Get("diskFullPaths").([]string)
 
 	if s.Skip {
 		log.Println("Skipping disk compaction step...")
 		return multistep.ActionContinue
 	}
 
-	ui.Say("Compacting the disk image")
-	if err := driver.CompactDisk(full_disk_path); err != nil {
-		state.Put("error", fmt.Errorf("Error compacting disk: %s", err))
-		return multistep.ActionHalt
-	}
-
-	if state.Get("additional_disk_paths") != nil {
-		if moreDisks := state.Get("additional_disk_paths").([]string); len(moreDisks) > 0 {
-			for i, path := range moreDisks {
-				ui.Say(fmt.Sprintf("Compacting additional disk image %d", i+1))
-				if err := driver.CompactDisk(path); err != nil {
-					state.Put("error", fmt.Errorf("Error compacting additional disk %d: %s", i+1, err))
-					return multistep.ActionHalt
-				}
-			}
+	ui.Say("Compacting all attached virtual disks...")
+	for i, diskPath := range diskPaths {
+		ui.Message(fmt.Sprintf("Compacting virtual disk %d", i+1))
+		// Get the file size of the virtual disk prior to compaction
+		fi, err := os.Stat(diskPath)
+		if err != nil {
+			state.Put("error", fmt.Errorf("Error getting virtual disk file info pre compaction: %s", err))
+			return multistep.ActionHalt
+		}
+		diskFileSizeStart := fi.Size()
+		// Defragment and compact the disk
+		if err := driver.CompactDisk(diskPath); err != nil {
+			state.Put("error", fmt.Errorf("Error compacting disk: %s", err))
+			return multistep.ActionHalt
+		}
+		// Get the file size of the virtual disk post compaction
+		fi, err = os.Stat(diskPath)
+		if err != nil {
+			state.Put("error", fmt.Errorf("Error getting virtual disk file info post compaction: %s", err))
+			return multistep.ActionHalt
+		}
+		diskFileSizeEnd := fi.Size()
+		// Report compaction results
+		log.Printf("Before compaction the disk file size was: %d", diskFileSizeStart)
+		log.Printf("After compaction the disk file size was: %d", diskFileSizeEnd)
+		percentChange := ((float64(diskFileSizeEnd) / float64(diskFileSizeStart)) * 100.0) - 100.0
```

Comment:
Is there ever a situation in the widlest stretch of possibility where diskFileSizeStart is 0?  I've caused too many crashes calculating percentages not to ask.

Reply:
I did think about adding some code to avoid any potential division by zero error. However, a vmdk will never (!) have a file size of zero - even when empty - as there is always a small amount of metadata stored internally.
That said, you've now made me doubt my logic; what if something went awry and the start size *was* zero; what if VMware change the vmdk format; the amount of code I would have to add to avoid a crash is tiny...

... I'll add the necessary code to avoid 😄